packages/jest-mock/src/index.ts

```diff
@@ -20,87 +20,91 @@ export type MockFunctionMetadataType =
 export type MockFunctionMetadata<
   T,
   Y extends Array<unknown>,
-  Type = MockFunctionMetadataType,
+  MetadataType = MockFunctionMetadataType,
 > = {
   ref?: number;
   members?: Record<string, MockFunctionMetadata<T, Y>>;
   mockImpl?: (...args: Y) => T;
   name?: string;
   refID?: number;
-  type?: Type;
+  type?: MetadataType;
   value?: T;
   length?: number;
 };
 
-export type MockableFunction = (...args: Array<any>) => any;
-export type MethodKeysOf<T> = {
-  [K in keyof T]: T[K] extends MockableFunction ? K : never;
-}[keyof T];
-export type PropertyKeysOf<T> = {
-  [K in keyof T]: T[K] extends MockableFunction ? never : K;
+export type ClassLike = {new (...args: Array<any>): any};
+
+export type FunctionLike = (...args: Array<any>) => any;
```

Comment:
These could work as `unknown` too, of course. In the other hand, `any` is nice type – just any JavaScript value. `unknown` is similar, but it is more about unknown type of user input (or unknown type of exception thrown), which has to be validated before consuming. There is no need to validate in this case, so I was thinking `any` is just the right type in these cases.

Reply:
I prefer `unknown`, since it _is_ unknown. If people want, they can cast to `any`, but I think we should be conservative on our side